src/TuDa.CIMS.Api/Repositories/PurchaseRepository.cs

```diff
@@ -0,0 +1,164 @@
+            }
+
+            existingItem.Buyer = buyer;
+        }
+
+        if (updateModel.Entries is not null)
+        {
+            var entries = await _context.PurchaseEntries
+                .Where(x => updateModel.Entries.Select(y => y.Id).Contains(x.Id)).ToListAsync();
+            if (entries.Count != updateModel.Entries.Count)
+            {
+                return Error.NotFound("Purchase.update", $"Some of the PurchaseEntries were not found.");
+            }
+
+            existingItem.Entries = entries;
+        }
+
+        existingItem.Buyer = updateModel.Buyer ?? existingItem.Buyer;
+        existingItem.Signature = updateModel.Signature ?? existingItem.Signature;
+        existingItem.Entries = updateModel.Entries ?? existingItem.Entries;
+        existingItem.CompletionDate = updateModel.CompletionDate ?? existingItem.CompletionDate;
+        existingItem.Completed = updateModel.Completed ?? existingItem.Completed;
+
+
+        await _context.SaveChangesAsync();
+
+        return new Updated();
+    }
+
+    /// <summary>
+    /// Removes a purchase with the specific ID from the database.
+    /// </summary>
+    /// <param name="id">the specific ID of the purchase</param>
+    /// <param name="workingGroupId">the specific ID of a workinggroup</param>
+    /// <returns></returns>
+    public async Task<ErrorOr<Deleted>> RemoveAsync(Guid id, Guid workingGroupId)
+    {
+        var itemToRemove = await _context
+            .WorkingGroups
+            .Where(i => i.Id == workingGroupId)
+            .SelectMany(i => i.Purchases)
+            .Where(i => i.Id == id)
+            .SingleOrDefaultAsync();
+
+        if (itemToRemove is null)
+        {
+            return Error.NotFound("Purchase.remove", $"Purchase with ID {id} was not found.");
+        }
+
+        _context.Purchases.Remove(itemToRemove);
+
+        await _context.SaveChangesAsync();
+        return Result.Deleted;
+    }
+
+    /// <summary>
+    /// Creates a new purchase in the database.
+    /// </summary>
+    /// <param name="workingGroupId">the specific ID of a workinggroup</param>
+    /// <param name="createModel">the model containing the updated values for the purchase</param>
+    /// <returns></returns>
+    public async Task<ErrorOr<Created>> CreateAsync(Guid workingGroupId, CreatePurchaseDto createModel)
+    {
+        var workingGroup = await _context.WorkingGroups.SingleOrDefaultAsync(r => r.Id == workingGroupId);
+        if (workingGroup is null)
+        {
+            return Error.NotFound("Purchase.create",
+                $"WorkingGroup with ID {workingGroupId} was not found.");
+        }
+
+        var newPurchase = new Purchase
+        {
+            Buyer = createModel.Buyer,
+            Signature = createModel.Signature,
+            CompletionDate = createModel.CompletionDate,
+            Entries = createModel.Entries,
+            Completed = createModel.Completed
```

Comment:
🔧 Check for the correlation of `CompletionDate` and `Complete`. CompletionDate is only not `null` if the `Purchase` is completed